### DISCUSSION TITLE
charon-cmd client connection on non standard port

### DISCUSSION THREAD

**Discussion by laloshifrin** [strongswan/strongswan#2254] [General]
Hi everyone,
I've been using Strongswan for at least a couple of years (maybe more) in roadwarrior configuration and it works great. Clients are almost all Linux working with NetworManager plugin enabled.
Now I have to connect with a Sailfish OS mobile system that uses Connman (as network manager) and Strongswan (self compiled) doesn't provide a plugin. I'm triyng to connect with charon-cmd but the server is listening on non standard ports. I tried, as documentation says, to change the default 4500 port adding "remote_port = XXXX" to swanctl.conf file but charon-cmd keeps trying to connect to 4500.
What am I doing wrong? Please help!!!
Thanks a lot for your work.

**Comment by tobiasbrunner** [comment]
[`charon-cmd`](https://docs.strongswan.org/docs/5.9/daemons/charon-cmd.html) is a stand-alone command line client that does not read settings from any config files (except strongswan.conf), they are all passed as command line arguments. However, there is currently no option to specify the server port. So unless you consider code changes, or modifying the destination port via firewall (e.g. DNAT in OUTPUT via iptables), you'd have to use the regular daemon and configure it via `swanctl.conf`.

**Reply by laloshifrin** [reply]
Everything works now. Thank you Tobias! A legend as always!
